provider/keycloak_openid_client.go

```diff
@@ -26,31 +31,112 @@ func resourceKeycloakOpenidClient() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"access_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(keycloakOpenidClientAccessTypes, false),
+			},
+			"client_secret": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"valid_redirect_uris": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+				DiffSuppressFunc: func(k, remoteRedirectUri, local string, data *schema.ResourceData) bool {
+					if remoteRedirectUri == "" || local == "" {
```

Comment:
I didn't dig deep into the Keycloak source code, but I'm wondering if the sort behavior of the `redirectUri` list is due to the fact that Keycloak represents the list internally as a [`Set<string>`](https://github.com/keycloak/keycloak/blob/b478472b3578b8980d7b5f1642e91e75d1e78d16/server-spi/src/main/java/org/keycloak/models/ClientModel.java#L75) (the concrete implementations usually being a `HashSet`).  There's technically no guaranteed iteration order of a `HashSet`, but it does seem consistently weird. Here's an example: https://ideone.com/ab52Y7 

https://stackoverflow.com/questions/18648521/how-is-this-hashset-producing-sorted-output

Reply:
The best I can guess here is that the keys are divided into buckets based on the hash code value and are inserted into their buckets in order of increasing hash code; the buckets are in the insertion order of the array that backs a `HashMap`.